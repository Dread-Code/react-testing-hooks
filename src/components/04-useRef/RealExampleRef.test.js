import React from "react";
import { shallow } from "enzyme";
import { RealExampleRef } from "./RealExampleRef";
import "@testing-library/jest-dom";

describe("RealExampleRef", () => {
  test("match to snapshoot", () => {
    const wrapper = shallow(<RealExampleRef />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Should match show the component", () => {
    const wrapper = shallow(<RealExampleRef />);
    wrapper.find("button").simulate("click");
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
