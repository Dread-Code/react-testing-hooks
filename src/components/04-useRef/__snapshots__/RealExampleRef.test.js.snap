// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RealExampleRef Should match show the component 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <MultipleCustomHooks />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;

exports[`RealExampleRef match to snapshoot 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
